azure-iot-device/azure/iot/device/provisioning/pipeline/pipeline_stages_provisioning.py

```diff
@@ -48,3 +63,331 @@ def _execute_op(self, op):
 
         else:
             self.send_op_down(op)
+
+
+class CommonProvisioningStage(PipelineStage):
+    # @pipeline_thread.runs_on_pipeline_thread
+    # def _get_retry_value(self, provisioning_op):
+    #     """
+    #     Return None if there is an error or else return the retry-after value obtained from the
+    #     service response. If the service has not given a retry-afer then return a default value.
+    #     The only reason a operation will not be retried is if it has moved to a different operation or else it has completed successfully.
+    #     In this case it means that a SendRegistrationRequestOperation
+    #     """
+    #     key_values_dict = provisioning_op.key_values
+    #     retry_after = (
+    #         None if "retry-after" not in key_values_dict else str(key_values_dict["retry-after"][0])
+    #     )
+    #     return retry_after
+
+    @pipeline_thread.runs_on_pipeline_thread
+    def _process_error(self, provisioning_op, prov_op_name, error):
```

Comment:
>_process_error [](start = 8, length = 14)

the word "process" doesn't tell me what this function does. Suggest naming it "_get_error_from_provisioning_response"

Reply:
renamed to only `_get_error_from_response` as it is evident from the stage name that it is meant for provisioning.

---
In reply to: [345903764](https://github.com/Azure/azure-iot-sdk-python/pull/391#discussion_r345903764) [](ancestors = 345903764)